### PR TITLE
feat: add self-join button and api to cases

### PR DESCRIPTION
**Pull request by wssheldon**
Screenshot 2024-07-02 at 10 28 45 AM

You can join dedicated case channels, if it is not closed. You cannot join non-dedicated case channels and there is no concept of 'subscribe'

### CODE REVIEW

File: src/dispatch/auth/permissions.py [hunk 1]
```diff
@@ -438,6 +438,20 @@ def has_required_permissions(
         return current_user.email in participant_emails
 
 
+class CaseJoinPermission(BasePermission):
+    def has_required_permissions(
+        self,
+        request: Request,
+    ) -> bool:
+        pk = PrimaryKeyModel(id=request.path_params["case_id"])
+        current_incident = incident_service.get(db_session=request.state.db, incident_id=pk.id)
```
Comment (mvilanova) [inline src/dispatch/auth/permissions.py:447]: Shouldn't we fetch the case not the incident here?

File: src/dispatch/case/views.py [hunk 2]
```diff
@@ -338,3 +339,25 @@ def delete_case(
                 }
             ],
         ) from None
+
+
+@router.post(
+    "/{case_id}/join",
+    summary="Adds an individual to an case.",
```
Comment (mvilanova) [inline src/dispatch/case/views.py:346]:
```suggestion
    summary="Adds an individual to a case.",
```

File: src/dispatch/case/views.py [hunk 3]
```diff
@@ -338,3 +339,25 @@ def delete_case(
...
+    dependencies=[Depends(PermissionsDependency([CaseJoinPermission]))],
+)
+def join_case(
```
Comment (mvilanova) [inline src/dispatch/case/views.py:349]: Lots of references to incident in this block

**Review by mvilanova — CHANGES_REQUESTED** [review]

**Comment by wssheldon** [comment]
@mvilanova Ready for review now if you'd like to take another look, I took a pass.

**Review by mvilanova — APPROVED** [review]

**Pull request merged**